pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -131,6 +131,7 @@ def on_evaluation_batch_start(self, batch: Any, batch_idx: int, dataloader_idx:
         model._current_dataloader_idx = dataloader_idx if num_dataloaders > 1 else None
 
         # track batch_size
+        assert self.trainer._results is not None
```

Comment:
Are the assert required for mypy ?

Reply:
Yeah, as `_results` can be None since the `PredictionLoop` don't have them... avoiding it would need a bit of refactoring